### PR TITLE
Add timeout to MIDI receive

### PR DESCRIPTION
**Pull request by ed-xmos**
Following a hang on MIDI test here http://srv-bri-jcim0:8080/blue/organizations/jenkins/XMOS%2Fsw_usb_audio/detail/PR-228/6/pipeline/250/

### CODE REVIEW

File: tests/test_midi.py [hunk 1]
```diff
@@ -56,6 +56,14 @@ def midi_duration(level, partial):
         duration = 10
     return duration
 
+def midi_receive_with_timeout(in_port, timeout_s=10):
+    for s in range(timeout_s):
+        msg = in_port.receive(block=False)
+        is msg is not None:
```
Comment (xross) [inline tests/test_midi.py:62]: If

Comment (xross) [reply tests/test_midi.py]: I quickly fixed this.

Comment (ed-xmos) [reply tests/test_midi.py]: Thanks! I was coming back to that and couldn't quite work out what had happened at first..

**Review by danielpieczko — APPROVED** [review]

File: Jenkinsfile [hunk 2]
```diff
@@ -385,7 +385,7 @@ pipeline {
                     }
 
                     withXTAG(["usb_audio_mc_xcai_dut", "usb_audio_mc_xcai_harness"]) { xtagIds ->
-                      sh "pytest -v --level ${params.TEST_LEVEL} --junitxml=pytest_result_windows11.xml \
+                      sh "pytest -s -v --level ${params.TEST_LEVEL} --junitxml=pytest_result_windows11.xml \
```
Comment (xross) [inline Jenkinsfile:388]: not sure if you meant to leave -s here or if it was just for debugging. Ignore if intentional.

**Review by xross — APPROVED** [review]

File: Jenkinsfile (hunk 2 above)
Comment (ed-xmos) [reply Jenkinsfile:388]: It was kind of intentional - You only look at the log if something has failed and there is precious little debug info so I thought it was helpful. Happy to remove if others prefer a cleaner output.

**Pull request merged**